SimpleRss/SimpleRss/ViewControllers/FeedViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  FeedViewController.swift
+//  SimpleRss
+//
+//  Created by Voldem on 11/14/18.
+//  Copyright © 2018 Vladimir Koptev. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class FeedViewController: UIViewController {
+    
+    var feedList = [Feed]()
+    var url: String!
+    
+    weak var tableView: UITableView!
+    
+    init(url: String!) {
+        super.init(nibName: nil, bundle: nil)
+        self.url = url
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let tableView = UITableView(frame: view.bounds)
+        
+        tableView.dataSource = self
+        tableView.delegate = self
+        
+        let nib = UINib(nibName: FeedViewCell.cellIdentifier(), bundle: nil)
+        tableView.register(nib, forCellReuseIdentifier: FeedViewCell.cellIdentifier())
+
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        
+        view.addSubview(tableView)
+        self.tableView = tableView;
+        
+        fetchXMLData()
+    }
+    
+    func fetchXMLData() {
+        XMLParserService().fetchXMLData(for: url) { (feedList, error) in
+            
+            if error == nil {
+                self.feedList = feedList!
+                self.tableView.reloadData()
+            }
+            else {
+                print(error?.localizedDescription ?? "Error")
+            }
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        self.title = "feed";
+    }
+}
+
+extension FeedViewController: UITableViewDataSource
+{
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: FeedViewCell.cellIdentifier()) as? FeedViewCell else { return UITableViewCell() }
+        
+        let feed = feedList[indexPath.row]
+        cell.TitleLabel.text = feed.title
+        cell.PreviewImageView?.downloaded(from: feed.picUrl)
+        cell.DescriptionLabel.text = feed.description
+     
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return feedList.count
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+}
+
+extension FeedViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath){
+        
+        guard let cell = tableView.cellForRow(at: indexPath) as? FeedViewCell else { return }
+        
+        cell.DescriptionLabel.numberOfLines = cell.DescriptionLabel.numberOfLines == 0 ? 1 : 0;
```

Comment:
Please, try the following case:
Open a feed, tap the first cell, scroll the table view.
How many expanded cell do you see?